cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1495,6 +1505,14 @@ def configure_apiserver(etcd_connection_string):
     else:
         remove_if_exists(audit_webhook_config_path)
 
+    if is_flag_set('layer.vault-kv.app-kv.set.encryption_key'):
+        # TODO: If Vault isn't available, it's probably still better to encrypt
+        # anyway and store the key in plaintext and leadership than to just
+        # give up on encryption entirely.
+        _write_encryption_config()
```

Comment:
What if this gets called before the `create_secure_storage` handler runs? My understanding is that this will go ahead and write the encryption key directly to the master's disk, which is something we otherwise seem to be avoiding. Seems like we wouldn't want to do that.

Reply:
Yeah, the `if is_flag_set` just above should have been on `kubernetes-master.secure-storage.created` instead.  I was originally going to support exposed config but realized that more work would be needed to support that so decided to defer it.  This got missed.  Thanks.

Reply:
Addressed.